crates/ntx-builder/src/builder/mod.rs

```diff
@@ -25,34 +64,57 @@ use crate::transaction::NtxError;
 /// against network accounts. These notes are identified and communicated by the block producer.
 /// The service maintains a list of unconsumed notes and periodically executes and proves
 /// transactions that consume them (reaching out to the store to retrieve state as necessary).
+///
+/// The builder manages the tasks for every network account on the chain through the coordinator.
 pub struct NetworkTransactionBuilder {
     /// Address of the store gRPC server.
-    pub store_url: Url,
+    store_url: Url,
     /// Address of the block producer gRPC server.
-    pub block_producer_url: Url,
+    block_producer_url: Url,
     /// Address of the remote prover. If `None`, transactions will be proven locally, which is
     /// undesirable due to the perofmrance impact.
-    pub tx_prover_url: Option<Url>,
+    tx_prover_url: Option<Url>,
     /// Interval for checking pending notes and executing network transactions.
-    pub ticker_interval: Duration,
+    ticker_interval: Duration,
     /// A checkpoint used to sync start-up process with the block-producer.
     ///
     /// This informs the block-producer when we have subscribed to mempool events and that it is
     /// safe to begin block-production.
-    pub bp_checkpoint: Arc<Barrier>,
+    bp_checkpoint: Arc<Barrier>,
+    /// Coordinator for managing actor tasks.
+    coordinator: Coordinator,
 }
 
 impl NetworkTransactionBuilder {
-    pub async fn serve_new(self) -> anyhow::Result<()> {
-        let store = StoreClient::new(self.store_url);
-        let block_producer = BlockProducerClient::new(self.block_producer_url);
+    /// Creates a new instance of the network transaction builder.
+    pub fn new(
+        store_url: Url,
+        block_producer_url: Url,
+        tx_prover_url: Option<Url>,
```

Comment:
could add a line on what happens if this is `None`?

Reply:
We have this passed through in various places via config, maybe we should create a more describing enum

```rust
enum Prover {
    Local,
    Delegated(Url),
}
```